test/spec/algoliasearch.helper/setQueryParameter.js

```diff
@@ -35,3 +35,28 @@ test('setChange should not change the current state: no real modification', func
   expect(changed).toBe(false);
   expect(helper.state).toBe(initialState);
 });
+
+test('setChange should ignore invalid userToken', function() {
+  var helper = algoliasearchHelper(fakeClient, null, {});
+  helper.setQueryParameter('userToken', null);
+  expect(helper.state.userToken).toBeUndefined();
+
+  helper.setQueryParameter('userToken', '');
+  expect(helper.state.userToken).toBeUndefined();
+
+  helper.setQueryParameter('userToken', 'my invalid token!');
+  expect(helper.state.userToken).toBeUndefined();
+});
+
+test('setChange should preserve valid userToken', function() {
```

Comment:
seems like this type exists in the whole file, but, the test should be called `setState`, not `setChange` :)

Reply:
https://github.com/algolia/algoliasearch-helper-js/pull/802/commits/c97cd22f9e3d43e3def417476084de50db91f941

I've put test cases in the all entries where userToken can be passed, to assure the warning message shows only once. What do you think? Too much?